ui/app/AppLayouts/Profile/views/wallet/ManageTokensView.qml

```diff
@@ -6,63 +6,123 @@ import StatusQ.Controls 0.1
 import shared.controls 1.0
 
 import AppLayouts.Profile.panels 1.0
+import AppLayouts.Wallet.panels 1.0
 
 ColumnLayout {
     id: root
 
     required property var sourcesOfTokensModel // Expected roles: key, name, updatedAt, source, version, tokensCount, image
     required property var tokensListModel // Expected roles: name, symbol, image, chainName, explorerUrl
 
+    required property var baseTokensModel
+    required property var baseCollectiblesModel
+
+    readonly property bool dirty: {
+        if (!loader.item)
+            return false
+        if (tabBar.currentIndex > 1)
+            return false
+        if (tabBar.currentIndex === 1 && (baseCollectiblesModel.isUpdating || baseCollectiblesModel.isFetching))
+            return false
+        return loader.item && loader.item.dirty
+    }
+
+    function saveChanges() {
+        if (tabBar.currentIndex > 1)
+            return
+        loader.item.saveSettings()
+    }
+
+    function resetChanges() {
+        if (tabBar.currentIndex > 1)
+            return
+        loader.item.revert()
+    }
+
+    QtObject {
+        id: d
+
+        function checkLoadMoreCollectibles() {
+            if (tabBar.currentIndex !== 1)
+                return
+            // If there is no more items to load or we're already fetching, return
+            if (!root.baseCollectiblesModel.hasMore || root.baseCollectiblesModel.isFetching)
+                return
+            root.baseCollectiblesModel.loadMore()
+        }
+    }
+
+    Connections {
+        target: root.baseCollectiblesModel
+        function onHasMoreChanged() {
+            d.checkLoadMoreCollectibles()
+        }
+        function onIsFetchingChanged() {
+            d.checkLoadMoreCollectibles()
+        }
+    }
+
     StatusTabBar {
         id: tabBar
 
         Layout.fillWidth: true
         Layout.topMargin: 5
 
         StatusTabButton {
-            id: assetsTab
+            leftPadding: 0
             width: implicitWidth
             text: qsTr("Assets")
         }
 
         StatusTabButton {
-            id: collectiblesTab
             width: implicitWidth
             text: qsTr("Collectibles ")
         }
 
         StatusTabButton {
-            id: tokensListTab
             width: implicitWidth
             text: qsTr("Token lists")
         }
     }
 
-    StackLayout {
-        id: stackLayout
-
+    // NB: we want to discard any pending unsaved changes when switching tabs or navigating away
+    Loader {
+        id: loader
         Layout.fillWidth: true
         Layout.fillHeight: true
-        currentIndex: tabBar.currentIndex
+        active: visible
+
+        sourceComponent: {
+            switch (tabBar.currentIndex) {
+            case 0:
```

Comment:
WDYT if we create an enum or `d` read-only properties to have a better readability in which tab we are working on and replace the hardcoded values in this file?

Reply:
Done